addons/ui/org.openhab.ui.cometvisu/src/main/java/org/openhab/ui/cometvisu/internal/util/ClientInstaller.java

```diff
@@ -0,0 +1,306 @@
+package org.openhab.ui.cometvisu.internal.util;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.ui.cometvisu.internal.Config;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Utility class for CometVisu client availability checks. It also provides methods to download
+ * the CometVisu client if it does not exist in the configured COMETVISU_WEBFOLDER
+ *
+ * @author Tobias Bräutigam - Initial contribution
+ *
+ */
+public class ClientInstaller {
+
+    static final Logger logger = LoggerFactory.getLogger(ClientInstaller.class);
+
+    /** the timeout to use for connecting to a given host (defaults to 5000 milliseconds) */
+    private int timeout = 5000;
+
+    /** URL for releases in github API */
+    private static final String releaseURL = "https://api.github.com/repos/CometVisu/CometVisu/releases";
+
+    private static final byte[] buffer = new byte[0xFFFF];
+
+    private Map latestRelease = null;
+
+    /** Regular expression to parse semver version strings */
+    private final Pattern semverPattern = Pattern.compile("[\\D]*([\\d]{1,3})\\.([\\d]{1,3})\\.([\\d]{1,3}).*");
+
+    private ArrayList<String> alreadyCheckedFolders = new ArrayList<String>();
```

Comment:
1. Please use the diamond operator, ie replace the second occurrence of the same generic by `<>`
2. Declare the type as `List<String>`